Tests/LegacyTestUser.php

```diff
@@ -6,7 +6,7 @@
 use Symfony\Component\Security\Core\User\AdvancedUserInterface;
 use Symfony\Component\Security\Core\User\UserInterface;
 
-class TestUser implements UserInterface, AdvancedUserInterface, LdapUserInterface
+class LegacyTestUser implements UserInterface, AdvancedUserInterface, LdapUserInterface
```

Comment:
The test for legacy code would fail with LdapUser because it implements only UserInterface and doesn't have a setPassword() method. That' why I have renamed the TestUser to LegacyTestUser